src/main/java/org/springframework/data/elasticsearch/core/mapping/ElasticsearchPersistentEntity.java

```diff
@@ -171,9 +171,18 @@ default ElasticsearchPersistentProperty getRequiredSeqNoPrimaryTermProperty() {
 
 	/**
 	 * returns the default settings for an index.
-	 * 
+	 *
 	 * @return settings as {@link Document}
 	 * @since 4.1
 	 */
 	Document getDefaultSettings();
+
+	/**
+	 * Resolves the routing for a bean.
+	 * 
+	 * @param bean the bean to resolve the routing for
+	 * @return routing value, may be {@literal null}
+	 */
+	@Nullable
+	String resolveRouting(Object bean);
```

Comment:
Actually, `bean` should be of type `T`. I think there's a similar constrain when obtaining `PersistentPropertyAccessor`. Interestingly, `isNew(…)` accepts `Object`.

Given the way how this method is used, it's fine to keep `Object` here and add a type assertion inside the implementation.

Reply:
assert to what type?

Reply:
That `bean` is instance of `T` (the entity type).